### PR TITLE
[Pages] Add zone settings on custom domains for debugging

### PR DESCRIPTION
**Pull request by KianNH**

### CODE REVIEW

File: content/pages/platform/debugging-pages.md [hunk 1]
```diff
@@ -72,6 +72,10 @@ Possible errors in this step could be caused by incorrect Pages Functions config
 
 If you are not using Functions or have reviewed that your Functions configuration does not contain any errors, review the [Cloudflare Status site](https://www.cloudflarestatus.com/) for Cloudflare network issues that could be causing the build failure. 
 
+## Differences between pages.dev and custom domains
+
+If your custom domain is proxied through Cloudflare, your zone's settings such as Auto Minify will apply. If Auto Minify is enabled for HTML, this will strip comments that some frameworks rely on such as Nuxt.js. If you're experiencing errors on your custom domain but not on your pages.dev domain, set the DNS record for your project to be DNS Only and see if the error persists. If not, review your zone's configuration to see what might cause it.
```
Comment (WalshyDev) [inline content/pages/platform/debugging-pages.md]:
```suggestion
If your custom domain is proxied through Cloudflare, your zone's settings such as Auto Minify and caching will apply.

If you are experiencing issues with a framework like Nuxt.js only on the custom domain, see if Auto Minify is enabled for HTML and disable it.

If you are experiencing issues with new content not being shown then check for a Page Rule with "Cache Everything" enabled, Pages handles its own cache.

If you're experiencing errors on your custom domain but not on your pages.dev domain, set the DNS record for your project to be DNS Only and see if the error persists. If not, review your zone's configuration to see what might cause it.
```

Comment (WalshyDev) [reply content/pages/platform/debugging-pages.md]: Not sure the best way to go through this but I'd love to talk about not just minification but also caching (and probably more in the future)

Comment (KianNH) [reply content/pages/platform/debugging-pages.md]: Yeah I didn't wanna go too in-depth but I also didn't wanna send them down tracking down 1 culprit setting amongst dozens

File: content/pages/platform/debugging-pages.md [hunk 2]
```diff
@@ -72,6 +72,16 @@ Possible errors in this step could be caused by incorrect Pages Functions config
 
 If you are not using Functions or have reviewed that your Functions configuration does not contain any errors, review the [Cloudflare Status site](https://www.cloudflarestatus.com/) for Cloudflare network issues that could be causing the build failure. 
 
+## Differences between pages.dev and custom domains
```
Comment (deadlypants1973) [inline content/pages/platform/debugging-pages.md]:
```suggestion
## Differences between `pages.dev` and custom domains
```

File: content/pages/platform/debugging-pages.md [hunk 3]
```diff
@@ -72,6 +72,16 @@ Possible errors in this step could be caused by incorrect Pages Functions config
...
+
+If your custom domain is proxied through Cloudflare, your zone's settings such as Auto Minify and caching will apply.
```
Comment (deadlypants1973) [inline content/pages/platform/debugging-pages.md]:
```suggestion
If your custom domain is proxied (orange-clouded) through Cloudflare, your zone's settings such as Auto Minify and caching will apply.
```

**Comment by KianNH** [comment]
Caching comment about the 'Cache Everything' page rule would also close https://github.com/cloudflare/cloudflare-docs/issues/4442

File: content/pages/platform/debugging-pages.md [hunk 4]
```diff
@@ -72,6 +72,16 @@ Possible errors in this step could be caused by incorrect Pages Functions config
...
+
+If you are experiencing issues with a framework like Nuxt.js only on the custom domain, see if Auto Minify is enabled for HTML and disable it.
```
Comment (deadlypants1973) [inline content/pages/platform/debugging-pages.md]:
```suggestion
If you are experiencing issues with a framework, like Nuxt.js, only on the custom domain, review if Auto Minify is enabled (log in to the [Cloudflare dashboard](https://dash.cloudflare.com/login) > **Speed** > **Optimization** > **Auto Minify**) for HTML and disable it.
```

File: content/pages/platform/debugging-pages.md [hunk 5]
```diff
@@ -72,6 +72,16 @@ Possible errors in this step could be caused by incorrect Pages Functions config
...
+
+If you are experiencing issues with new content not being shown then check for a Page Rule with "Cache Everything" enabled, Pages handles its own cache.
```
Comment (deadlypants1973) [inline content/pages/platform/debugging-pages.md]:
```suggestion
If you are experiencing issues with new content not being shown, go to **Rules** > **Page Rules** in the Cloudfalre dashboard and check for a Page Rule with **Cache Everything** enabled. Pages handles its own cache.
```
Is this saying to disable the Page Rule? Or leave it enabled?

File: content/pages/platform/debugging-pages.md [hunk 6]
```diff
@@ -72,6 +72,16 @@ Possible errors in this step could be caused by incorrect Pages Functions config
...
+
+If you're experiencing errors on your custom domain but not on your pages.dev domain, set the DNS record for your project to be DNS Only and see if the error persists. If not, review your zone's configuration to see what might cause it.
```
Comment (deadlypants1973) [inline content/pages/platform/debugging-pages.md]:
```suggestion
If you are experiencing errors on your custom domain but not on your `pages.dev` domain, go to **DNS** in the Cloudflare dashboard and set the DNS record for your project to be **DNS Only** (grey cloud). If the error persists, review your zone's configuration.
```

File: content/pages/platform/debugging-pages.md (hunk 5 above)
Comment (KianNH) [reply content/pages/platform/debugging-pages.md]: We'd want the rule removing - it causes 'another' layer of caching on top of the one that Pages already has but Pages isn't going to automatically invalidate that cache on deploy so people end up with old content.

Comment (KianNH) [reply content/pages/platform/debugging-pages.md]:
```suggestion
If you are experiencing issues with new content not being shown, go to **Rules** > **Page Rules** in the Cloudflare dashboard and check for a Page Rule with **Cache Everything** enabled. If present, remove this rule as Pages handles its own cache.
```

**Review by deadlypants1973 — APPROVED** [review]

**Pull request merged**